.dagger/test.go

```diff
@@ -204,6 +205,70 @@ func (t *Test) test(
 		false, // -update
 		testVerbose,
 	).Sync(ctx)
+	if err != nil {
+		return err
+	}
+
+	return t.publishTestResults(ctx, ctr)
+}
+
+func (t *Test) publishTestResults(ctx context.Context, ctr *dagger.Container) error {
```

Comment:
Can we split this publishing up to the dashboard site into a separate PR?

I'm a little worried about exporting our test results to an external service that can go down that we don't have full control over. I think we need more discussion on this. If we *do* do this, I don't think it should be part of every `test` call, which is also used locally - it would need to have logic to only apply in CI.

The rest of the PR around `gotestsum`/etc looks good, and I don't want to block those changes from getting in while we're discussing that.

Reply:
apologies for the noise. But this endpoint is temporary here. Me and @gerhard had discussion regarding this and we are going to deploy this service on dagger infrastructure. This is me just testing out stuff and making sure other things are aligned and work.

Reply:
I just changed this PR back to being a draft PR until I am done testing other things.